From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct xfrm_dst
Patch-mainline: never, kabi
References: kabi

In 4.4.76, commit 398ac7a19f17386d17f563ebfe273ced37c4897a (xfrm: fix
stack access out of bounds with CONFIG_XFRM_SUB_POLICY), upstream commit
9b3eb54106cf6acd03f07cf0ab01c13676a226c2 removed two members from struct
xfrm_dst. This indeed changed the layout of the structure and the kABI
checker now complains.

Just reintroduce the members back and be done with it.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/xfrm.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/net/xfrm.h
+++ b/include/net/xfrm.h
@@ -948,6 +948,10 @@ struct xfrm_dst {
 	struct flow_cache_object flo;
 	struct xfrm_policy *pols[XFRM_POLICY_TYPE_MAX];
 	int num_pols, num_xfrms;
+#ifdef CONFIG_XFRM_SUB_POLICY
+	struct flowi *origin; /* unused -- kABI */
+	struct xfrm_selector *partner; /* unused -- kABI */
+#endif
 	u32 xfrm_genid;
 	u32 policy_genid;
 	u32 route_mtu_cached;
